core/trino-main/src/main/java/io/trino/testing/PlanTester.java

```diff
@@ -302,15 +303,20 @@ public class PlanTester
 
     public static PlanTester create(Session defaultSession)
     {
-        return new PlanTester(defaultSession, 1);
+        return new PlanTester(defaultSession, identity(), 1);
     }
 
     public static PlanTester create(Session defaultSession, int nodeCountForStats)
     {
-        return new PlanTester(defaultSession, nodeCountForStats);
+        return new PlanTester(defaultSession, identity(), nodeCountForStats);
     }
 
-    private PlanTester(Session defaultSession, int nodeCountForStats)
+    public static PlanTester create(Session defaultSession, Function<Metadata, Metadata> metadataDecorator, int nodeCountForStats)
+    {
+        return new PlanTester(defaultSession, metadataDecorator, nodeCountForStats);
+    }
+
+    private PlanTester(Session defaultSession, Function<Metadata, Metadata> metadataDecorator, int nodeCountForStats)
```

Comment:
instead of `Function<Metadata, Metadata> metadataDecorator`, you can take `Optional<Metadata> metadata` as parameter